pkg/sign/keys.go

```diff
@@ -39,8 +40,8 @@ type Keypair interface {
 
 type EphemeralKeypairOptions struct {
 	// Optional hint of for signing key
-	Hint []byte
-	// TODO: support additional key algorithms
+	Hint          []byte
+	HashAlgorithm protocommon.HashAlgorithm
```

Comment:
Ah, yeah, we can support some additional keys in https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_common.proto#L62, especially for key types and hashes we expect to be very common.

But I don't think that `pkg/sign/keys.go` should implement every possible combination. The point of the `Keypair` interface is so that `sigstore-go` doesn't have to implement every key type - you can provide your own implementation to that interface.

That's why I was curious what @alexiscampan's use-case is. If they were anticipating lots of clients needing to sign with a specific key type and hash, or if it would make more sense for them to implement their own `Keypair` when they use `sigstore-go`.

Reply:
That makes sense for KMS keys or other key interfaces that pull in dependencies or experimental algorithms, e.g. PQC. Or for signing where keys are managed, e.g on disk or HSMs.

For algorithms that are natively supported in Go and where keys are used in ephemeral signing, why not support them here too? The default of ECDSA-P256 is reasonable, but in the spirit of cryptographic agility (https://github.com/sigstore/sigstore-go/issues/74), I would like to add in a way to configure RSA, ed25519, and other elliptic curves. I recognize this is a case of YAGNI, but it's straightforward to add this support and removes a barrier for potential integrators while still directing them to the default, preferred Sigstore signing workflow.

Reply:
> Ah, yeah, we can support some additional keys in https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_common.proto#L62, especially for key types and hashes we expect to be very common.
> 
> But I don't think that `pkg/sign/keys.go` should implement every possible combination. The point of the `Keypair` interface is so that `sigstore-go` doesn't have to implement every key type - you can provide your own implementation to that interface.
> 
> That's why I was curious what @alexiscampan's use-case is. If they were anticipating lots of clients needing to sign with a specific key type and hash, or if it would make more sense for them to implement their own `Keypair` when they use `sigstore-go`.

Hey @steiza, I have no particular use case for this specific feature. With my company we are going to need the signing part, we could use other librairies like the one in python to sign our images but all our applications are written in Go so we wanted to stay consistant. I also think that the default ECDSA-P256 is sufficient for our use case. I just wanted to do a first contribution before contributing on the signing part. I will close this PR for now as the priority of this feature is not high and we don't really know if it will be useful. I will try to give some support on the signing part which is the main feature of this library. Thanks for your fast review.